test_infra/stacks/cleanrooms_stack.py

```diff
@@ -0,0 +1,241 @@
+
+
+class CleanRoomsStack(Stack):  # type: ignore
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        **kwargs: str,
+    ) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        self.collaboration = cleanrooms.CfnCollaboration(
+            self,
+            "Collaboration",
+            name="AWS SDK for pandas - Testing",
+            creator_display_name="Collaborator Creator",
+            creator_member_abilities=["CAN_QUERY", "CAN_RECEIVE_RESULTS"],
+            description="Collaboration Room for AWS SDK for pandas test infrastructure",
+            members=[],
+            query_log_status="ENABLED",
+        )
+
+        self.membership = cleanrooms.CfnMembership(
+            self,
+            "Membership",
+            collaboration_identifier=self.collaboration.attr_collaboration_identifier,
+            query_log_status="ENABLED",
+        )
+
+        self.cleanrooms_service_role = iam.Role(
+            self,
+            "Service Role",
+            assumed_by=iam.CompositePrincipal(
+                iam.ServicePrincipal("cleanrooms.amazonaws.com").with_conditions(
+                    {
+                        "StringLike": {
+                            "sts:ExternalId": f"arn:aws:*:{self.region}:*:dbuser:*/{self.membership.attr_membership_identifier}*"
+                        }
+                    }
+                ),
+                iam.ServicePrincipal("cleanrooms.amazonaws.com").with_conditions(
+                    {
+                        "ForAnyValue:ArnEquals": {
+                            "aws:SourceArn": f"arn:aws:cleanrooms:{self.region}:{self.account}:membership/{self.membership.attr_membership_identifier}"
+                        }
+                    }
+                ),
+            ),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name("service-role/AWSGlueServiceRole"),
+                iam.ManagedPolicy.from_aws_managed_policy_name("AmazonS3ReadOnlyAccess"),
+            ],
+        )
+
+        self.bucket = s3.Bucket(
+            self,
+            "Bucket",
+            block_public_access=s3.BlockPublicAccess(
+                block_public_acls=True,
+                block_public_policy=True,
+                ignore_public_acls=True,
+                restrict_public_buckets=True,
+            ),
+            lifecycle_rules=[
+                s3.LifecycleRule(
+                    id="CleaningUp",
+                    enabled=True,
+                    expiration=Duration.days(1),
+                    abort_incomplete_multipart_upload_after=Duration.days(1),
+                ),
+            ],
+            versioned=True,
+        )
+
+        self.database = glue.Database(
+            self,
+            id="Glue Database",
+            database_name="aws_sdk_pandas_cleanrooms",
+            location_uri=f"s3://{self.bucket.bucket_name}",
+        )
```

Comment:
A new S3 bucket and Glue database are required because AWS Clean Rooms does not currently support Amazon S3 bucket locations that are registered with AWS Lake Formation. It's annoying but couldn't think of a better approach

Reply:
Should we be referencing this in our docs somewhere? In the docstring for the method under the bucket param maybe?

And agreed I think this is the best approach for testing for now.

Reply:
Oh it would only be in our test infra, so no need for docs as not customer facing. Unless you are referring to something else?